src/sql/query.rs

```diff
@@ -1836,6 +1836,52 @@ fn plan_function<'a>(
                 Ok(expr)
             }
 
+            "round" => {
+                // https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_round
+                // The return value has the same type as the first argument
+                // (assuming that it is integer, double, or decimal).
+                // This means that for an integer argument, the result is an integer (no decimal places)
+                if sql_func.args.is_empty() || sql_func.args.len() > 2 {
+                    bail!(
+                        "round expects 1 or 2 arguments, got {}",
+                        sql_func.args.len()
+                    );
+                }
+
+                let expr1 = plan_expr(ecx, &sql_func.args[0], None)?;
+                let typ1 = ecx.column_type(&expr1);
+
+                if sql_func.args.len() == 1 {
+                    let expr = match typ1.scalar_type {
+                        ScalarType::Int32 => expr1.call_unary(UnaryFunc::RoundInt32),
+                        ScalarType::Int64 => expr1.call_unary(UnaryFunc::RoundInt64),
```

Comment:
For PostgreSQL compatibility, we don't actually want overloads for int32/int64. Instead we want to call `promote_number_floatdec` to promote the input to either a float or a decimal, and then we just need implementations for floats and decimals.